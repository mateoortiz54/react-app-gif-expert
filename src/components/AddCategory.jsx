import {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory})=> {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) => {
        setInputValue(target.value)
    }

    const onSubmit = (event)=>{
        event.preventDefault();

        const inputV = inputValue.trim()
        // trim es para limpiar el elemento al final y al principio de espacios
        if (inputV.length <=1 ) return;

        onNewCategory(inputV)
        setInputValue('')
    }

    return (
        // <form onSubmit={(event) => onSubmit(event)}>    
        <form onSubmit={onSubmit} aria-label="form">    
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                // onChange={(event)=>onInputChange(event)}
                onChange={onInputChange} />
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
