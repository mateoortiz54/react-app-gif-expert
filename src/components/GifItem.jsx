

export const GifItem = (props) => {

    return (
        <div>
            <img src={props.url} alt={props.title} /> 
            <p>{props.title}</p>
        </div>
    )
}

