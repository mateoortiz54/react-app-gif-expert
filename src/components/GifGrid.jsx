import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifItem } from './GifItem'


export const GifGrid = ({category}) => {

    const {images, isLoading} = useFetchGifs(category)

    return (
        <div>
            
            <h3>{category}</h3>
            {
                isLoading && (<h2>Cargando imagenes....</h2>)
            }
            { <div className='card-grid'>
            {
                images.map(obj=>(
                    
                    <GifItem
                        key={obj.id}
                        { ...obj }
                    />
                ))
            }
            </div>}
            
        </div>
    )


}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}