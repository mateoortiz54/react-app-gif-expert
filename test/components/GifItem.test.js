import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en GifItem', () => {

    const title = 'saitama'
    const url = 'https://one-punch-man/saitama.jpg'

    test('el titulo y el url deben de ser obligatorios', () => {
        

        const  {container} = render(<GifItem title={title} url={url}/>)

        expect(container).toMatchSnapshot()
        
    })
    test('debe de mostrar la imagen con el url y alt indicado', () => {

        const  {container} = render(<GifItem title={title} url={url}/>)
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)

    });
    test('debe de mostrar el titulo en el componente', () => {

        const  {container} = render(<GifItem title={title} url={url}/>);
        expect(screen.getByAltText(title)).toBeTruthy();
    });
})