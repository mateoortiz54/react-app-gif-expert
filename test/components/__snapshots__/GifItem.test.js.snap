// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem el titulo y el url deben de ser obligatorios 1`] = `
<div>
  <div>
    <img
      alt="saitama"
      src="https://one-punch-man/saitama.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
